azurerm/resource_arm_storage_account.go

```diff
@@ -701,7 +701,7 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 		}
 
 		networkRules := props.NetworkRuleSet
-		if networkRules != nil && len(*networkRules.IPRules) > 0 && len(*networkRules.VirtualNetworkRules) > 0 {
+		if networkRules != nil {
```

Comment:
we should be able to remove this `if` statement entirely and replace it with ones inside the flatten functions? this means we'll set the fields to an empty value if nil is returned from the api?